analyses/doublet-detection/scripts/01b_run-scrublet.py

```diff
@@ -21,11 +21,20 @@ def run_scrublet(adata: anndata.AnnData, random_seed: int) -> pandas.DataFrame:
     """
 
     scrub = scrublet.Scrublet(adata.X, random_state = random_seed)
-    # predicted_doublets is boolean array, where True are predicted doublets and False are predicted singlets
-    doublet_scores, predicted_doublets = scrub.scrub_doublets()
 
-    # convert True/False to string values
-    predicted_doublets_str = [("doublet" if x else "singlet") for x in predicted_doublets]
+    # Predict doublets if possible. If it fails, return all NAs
+    try:
+        # predicted_doublets is boolean array, where True are predicted doublets and False are predicted singlets
+        doublet_scores, predicted_doublets = scrub.scrub_doublets()
+        # convert True/False to string values
+        predicted_doublets_str = [("doublet" if x else "singlet") for x in predicted_doublets]
+    except:
+        print(
+            "Could not run scrublet on the provided dataset."
+            "The output TSV file will contain only NA values."
+        )
+        doublet_scores = ["NA"]*len(adata.obs_names)
+        predicted_doublets_str = ["NA"]*len(adata.obs_names)
```

Comment:
Review question: is there another preferred way here to get lists of `"NA"` strings?

Reply:
Do you need it to be a string? Or would it be better to use `pandas.NA` here? Either way, you should be able to use a single value for these, as the number of rows  of the data frame will be determined by the barcodes and pandas will recycle single values as needed. 

(You may want to use `numpy.nan` for the scores to have them be floats, but since you are just outputting a file and probably don't care too much about the types as you would if you were going to process the data frame itself.)

Reply:
I think this is fine, but if you wanted to load in `numpy` then you could just do: 
`numpy.full(shape=len(adata.obs_names), fill_value ="NA")`. 

I think either one is okay.

Reply:
Just saw Josh's suggestion and I like that better. If you just need a single value then use that.

Reply:
> but since you are just outputting a file and probably don't care too much about the types as you would if you were going to process the data frame itself.)

this is definitely the thought I had that me think `"NA"` as a plain string was good enough! but yeah a single value seems better